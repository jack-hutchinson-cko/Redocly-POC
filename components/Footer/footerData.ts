export const dataFooter = {
	body: [
		{
			slice_type: 'links_section',
			slice_label: null,
			items: [
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Solution overview',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/solutions',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Unified payments platform',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/solutions/unified-api',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'International Coverage',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/solutions/international-coverage',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Data and Reporting',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/solutions/data-and-reporting',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Integrations',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/solutions/integration',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Fraud protection',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/solutions/fraud',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Payment methods',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/payment-methods',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Pricing',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/pricing',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Partners',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/partners',
							spans: [],
						},
					],
				},
			],
			primary: {
				title: [
					{
						type: 'heading2',
						text: 'Solution',
						spans: [],
					},
				],
			},
		},
		{
			slice_type: 'links_section',
			slice_label: null,
			items: [
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'About us',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/about-us',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Careers',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/careers',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Newsroom',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/newsroom',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Contact us',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/contact-us',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Become a partner',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/partners/join',
							spans: [],
						},
					],
				},
			],
			primary: {
				title: [
					{
						type: 'heading2',
						text: 'Company',
						spans: [],
					},
				],
			},
		},
		{
			slice_type: 'links_section',
			slice_label: null,
			items: [
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Blog',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/blog',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Customer stories',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/case-studies',
							spans: [],
						},
					],
				},
			],
			primary: {
				title: [
					{
						type: 'heading2',
						text: 'Insights',
						spans: [],
					},
				],
			},
		},
		{
			slice_type: 'links_section',
			slice_label: null,
			items: [
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Documentation',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: 'https://docs.checkout.com/',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'API Reference',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: 'https://api-reference.checkout.com/',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Support',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: 'mailto:support@checkout.com',
							spans: [],
						},
					],
				},
			],
			primary: {
				title: [
					{
						type: 'heading2',
						text: 'Resources',
						spans: [],
					},
				],
			},
		},
	],
	title: [{ title: 'mock data' }],
};
