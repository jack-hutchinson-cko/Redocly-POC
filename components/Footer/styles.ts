import styled from 'styled-components';
import { colours, device } from '../css/tokens';

export const Background = styled.div`
	height: 100%;
	width: 100%;
	background: ${colours.S1000};
	margin-top: 0;
`;

export const Container = styled.div``;

export const LogoContainer = styled.div`
	height: 50px;
	display: flex;
	align-items: center;

	select {
		flex: 0;
	}
`;

export const LogoWrapper = styled.div`
	flex: 1;
`;

export const LineSeparator = styled.div`
	width: 100%;
	height: 1px;
	max-width: 1232px;
	background: ${colours.S1000};
	margin: 40px auto;
`;

export const Section = styled.div`
	position: relative;
	margin: 0 auto;
	padding: 64px 0px;
	max-width: 1360px;
	width: 100%;
	background-color: ${colours.S1000};

	${device.toMobileXL} {
		padding: 64px 24px;
	}
`;
