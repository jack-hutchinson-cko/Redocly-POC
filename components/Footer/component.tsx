import React from 'react';
import { Section } from './styles';
import { Background, LogoContainer, LogoWrapper, LineSeparator } from './styles';
import { CheckoutLogo } from '../svg/CheckoutLogo';
import FooterLinkList from './FooterLinkList';
import SubFooter from './SubFooter';
import { dataFooter } from './footerData';

const Footer = () => (
	<Background>
		<Section>
			<LogoContainer>
				<LogoWrapper>
					<CheckoutLogo />
				</LogoWrapper>
			</LogoContainer>

			<FooterLinkList links={dataFooter.body} />

			<LineSeparator />

			<SubFooter />
		</Section>
	</Background>
);

export default Footer;
