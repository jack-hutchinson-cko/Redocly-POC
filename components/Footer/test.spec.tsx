import { shallow } from 'enzyme';
import React from 'react';
import Footer from './component';

describe('Footer component', () => {
	test('component matches snapshot', () => {
		expect.assertions(1);

		const wrapper = shallow(<Footer />);

		expect(wrapper).toMatchSnapshot();
	});
});
