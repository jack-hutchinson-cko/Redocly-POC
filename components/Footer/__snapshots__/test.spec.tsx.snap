// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component component matches snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        <CheckoutLogo />
      </styled.div>
    </styled.div>
    <FooterLinkList
      links={
        Array [
          Object {
            "items": Array [
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/solutions",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Solution overview",
                    "type": "paragraph",
                  },
                ],
              },
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/solutions/unified-api",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Unified payments platform",
                    "type": "paragraph",
                  },
                ],
              },
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/solutions/international-coverage",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "International Coverage",
                    "type": "paragraph",
                  },
                ],
              },
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/solutions/data-and-reporting",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Data and Reporting",
                    "type": "paragraph",
                  },
                ],
              },
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/solutions/integration",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Integrations",
                    "type": "paragraph",
                  },
                ],
              },
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/solutions/fraud",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Fraud protection",
                    "type": "paragraph",
                  },
                ],
              },
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/payment-methods",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Payment methods",
                    "type": "paragraph",
                  },
                ],
              },
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/pricing",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Pricing",
                    "type": "paragraph",
                  },
                ],
              },
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/partners",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Partners",
                    "type": "paragraph",
                  },
                ],
              },
            ],
            "primary": Object {
              "title": Array [
                Object {
                  "spans": Array [],
                  "text": "Solution",
                  "type": "heading2",
                },
              ],
            },
            "slice_label": null,
            "slice_type": "links_section",
          },
          Object {
            "items": Array [
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/about-us",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "About us",
                    "type": "paragraph",
                  },
                ],
              },
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/careers",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Careers",
                    "type": "paragraph",
                  },
                ],
              },
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/newsroom",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Newsroom",
                    "type": "paragraph",
                  },
                ],
              },
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/contact-us",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Contact us",
                    "type": "paragraph",
                  },
                ],
              },
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/partners/join",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Become a partner",
                    "type": "paragraph",
                  },
                ],
              },
            ],
            "primary": Object {
              "title": Array [
                Object {
                  "spans": Array [],
                  "text": "Company",
                  "type": "heading2",
                },
              ],
            },
            "slice_label": null,
            "slice_type": "links_section",
          },
          Object {
            "items": Array [
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/blog",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Blog",
                    "type": "paragraph",
                  },
                ],
              },
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "/case-studies",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Customer stories",
                    "type": "paragraph",
                  },
                ],
              },
            ],
            "primary": Object {
              "title": Array [
                Object {
                  "spans": Array [],
                  "text": "Insights",
                  "type": "heading2",
                },
              ],
            },
            "slice_label": null,
            "slice_type": "links_section",
          },
          Object {
            "items": Array [
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "https://docs.checkout.com/",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Documentation",
                    "type": "paragraph",
                  },
                ],
              },
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "https://api-reference.checkout.com/",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "API Reference",
                    "type": "paragraph",
                  },
                ],
              },
              Object {
                "link_href": Array [
                  Object {
                    "spans": Array [],
                    "text": "mailto:support@checkout.com",
                    "type": "paragraph",
                  },
                ],
                "link_name": Array [
                  Object {
                    "spans": Array [],
                    "text": "Support",
                    "type": "paragraph",
                  },
                ],
              },
            ],
            "primary": Object {
              "title": Array [
                Object {
                  "spans": Array [],
                  "text": "Resources",
                  "type": "heading2",
                },
              ],
            },
            "slice_label": null,
            "slice_type": "links_section",
          },
        ]
      }
    />
    <styled.div />
    <SubFooter />
  </styled.div>
</styled.div>
`;
